src/__experimental__/components/checkbox/checkbox-group.component.js

```diff
@@ -2,49 +2,50 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import tagComponent from '../../../utils/helpers/tags';
 import { StyledCheckboxGroup } from './checkbox.style';
-import { withValidation } from '../../../components/validations';
 import FormField from '../form-field';
 
 const CheckboxGroup = (props) => {
   const {
     children,
-    groupName,
-    hasError,
-    hasWarning,
-    hasInfo
+    id,
+    value,
+    defaultValue
   } = props;
-
-  const groupLabelId = `${groupName}-label`;
+  const defaultCheckedValue = (defaultValue !== undefined) ? defaultValue : [];
+  const checkboxGroupValue = (value !== undefined) ? value : defaultCheckedValue;
+  const groupLabelId = `${id}-label`;
 
   const buttons = React.Children.map(children, (child) => {
+    let checked = false;
+    let defaultChecked = false;
+
+    if (checkboxGroupValue.length > 0) {
+      checked = (checkboxGroupValue.indexOf(child.props.value) !== -1);
+    }
+
+    if (defaultCheckedValue.length > 0) {
+      defaultChecked = (defaultCheckedValue.indexOf(child.props.value) !== -1);
+    }
+
     const handleChange = (ev) => {
       child.props.onChange(ev);
     };
 
-    let childProps = {
-      inputName: groupName,
-      onChange: handleChange
+    const childProps = {
+      inputName: id,
+      onChange: handleChange,
+      ...(checked && { checked }),
+      ...(defaultChecked && { defaultChecked })
     };
 
-    if (!child.props.checked) {
-      childProps = {
-        ...childProps,
-        hasError,
-        hasWarning,
-        hasInfo
-      };
-    }
-
     return React.cloneElement(child, childProps);
   });
 
   return (
     <StyledCheckboxGroup
       aria-labelledby={ groupLabelId }
       role='checkbox'
-      hasError={ hasError }
-      hasWarning={ hasWarning }
-      hasInfo={ hasInfo }
+      value={ checkboxGroupValue }
```

Comment:
Why does the styled component need the value?

Reply:
I don't think it does. `<StyledCheckboxGroup>` in `src/__experimental__/components/checkbox/checkbox.style.js` doesn't seem to use `value`. Removed in d418bfc